arch/risc-v/src/common/riscv_pmp.c

```diff
@@ -100,32 +101,6 @@ typedef struct pmp_entry_s pmp_entry_t;
  * Private Functions
  ****************************************************************************/
 
-/****************************************************************************
- * Name: log2ceil
- *
- * Description:
- *   Calculate the up-rounded power-of-two for input.
- *
- * Input Parameters:
- *   size - The size of the PMP region.
- *
- * Returned Value:
- *   Power-of-two for argument, rounded up.
- *
- ****************************************************************************/
-
-static uintptr_t log2ceil(uintptr_t size)
-{
-  uintptr_t pot = 0;
-
-  for (size = size - 1; size > 1; size >>= 1)
-    {
-      pot++;
-    }
-
-  return pot;
-}
-
```

Comment:
@eenurkka maybe this log2ceil could be moved to libdsp or other place to be reused for other arches that don't have it in binutils (older version of binutils).

BTW did you see that: https://stackoverflow.com/questions/15043483/faster-log2ceil

There is a faster implementation of log2ceil that just use logic shift and AND operation

Reply:
is this enough?
https://github.com/apache/nuttx/blob/master/include/nuttx/lib/math32.h#L41-L60

Reply:
in 'mu0` you have a bit_floor and bit_ceil implementation macthing c++. `__mu0_bit_floor_u8__` is performed on `unsigned long long` size of `uintptr_t` must be pounded defined i.e arch32 vs arch64.

https://github.com/mu578/mu0
https://en.cppreference.com/w/cpp/numeric/bit_floor
https://en.cppreference.com/w/cpp/numeric/bit_ceil